sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs

```diff
@@ -652,6 +823,50 @@ protected internal virtual BlobClient WithClientSideEncryptionOptionsCore(Client
                 conditions: overwrite ? null : new BlobRequestConditions { IfNoneMatch = new ETag(Constants.Wildcard) },
                 cancellationToken: cancellationToken);
 
+        /// <summary>
+        /// The <see cref="UploadAsync(BinaryData, bool, CancellationToken)"/> operation
+        /// creates a new block blob or throws if the blob already exists.
+        /// Setting <paramref name="overwrite"/> to true allows updating the
+        /// content of an existing block blob.  Updating an existing block blob
+        /// overwrites any existing metadata on the blob.
+        ///
+        /// For partial block blob updates and other advanced features, please
+        /// see <see cref="BlockBlobClient"/>.  To create or modify page or
+        /// append blobs, please see <see cref="PageBlobClient"/> or
+        /// <see cref="AppendBlobClient"/>.
```

Comment:
We should also think about adding BinaryData support to the specialized blobs in the future.  I could see leaving Block and Page uploading with just Streams because both of those are highly specialized, but I could see the use in `AppendBlobClient.Append(BinaryData)` for customers to easily add a string of text.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/19273